lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -150,106 +133,255 @@ extension ASTGenVisitor {
 // Misc visits.
 // TODO: Some of these are called within a single file/method; we may want to move them to the respective files.
 extension ASTGenVisitor {
-  public func visit(_ node: MemberBlockItemSyntax) -> ASTNode {
-    visit(Syntax(node.decl))
+  public func generate(_ node: MemberBlockItemSyntax) -> ASTNode {
+    generate(node.decl)
   }
 
-  public func visit(_ node: InitializerClauseSyntax) -> ASTNode {
-    visit(node.value)
+  public func generate(_ node: InitializerClauseSyntax) -> ASTNode {
+    generate(node.value)
   }
 
-  public func visit(_ node: ConditionElementSyntax) -> ASTNode {
-    visit(node.condition)
+  public func generate(_ node: ConditionElementSyntax) -> ASTNode {
+    generate(node.condition)
+  }
+
+  public func generate(_ node: CodeBlockItemSyntax) -> ASTNode {
+    generate(node.item)
+  }
+
+  public func generate(_ node: ArrayElementSyntax) -> ASTNode {
+    generate(node.expression)
+  }
+    
+  func generate(_ node: DeclSyntax) -> ASTNode {
+      return generate(Syntax(node))
   }
 
-  public func visit(_ node: CodeBlockItemSyntax) -> ASTNode {
-    visit(node.item)
+  func generate(_ node: ExprSyntax) -> ASTNode {
+    return generate(Syntax(node))
   }
 
-  public func visit(_ node: ArrayElementSyntax) -> ASTNode {
-    visit(node.expression)
+  func generate(_ node: PatternSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: StmtSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: TypeSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: some SyntaxChildChoices) -> ASTNode {
+    return self.generate(Syntax(node))
   }
 
   @inline(__always)
-  func visit(_ node: CodeBlockItemListSyntax) -> BridgedArrayRef {
-    node.lazy.map { self.visit($0).bridged }.bridgedArray(in: self)
+  func generate(_ node: CodeBlockItemListSyntax) -> BridgedArrayRef {
+    node.lazy.map { self.generate($0).bridged }.bridgedArray(in: self)
+  }
+    
+  public func generate(_ node: Syntax) -> ASTNode {
```

Comment:
Could we split off the new top-level `generate` methods that take `Syntax`, `{Decl,Stmt,Pattern,Expr,Type}Syntax, and `some SyntaxChildChoices` into their own extension? This extension is sort of a dumping ground for miscellanous `generate` method such as `func visit(_ node: ArrayElementSyntax)` that we haven’t yet found a better place for.